CMakeLists.txt

```diff
@@ -207,522 +79,35 @@ if(APPIMAGE)
-		INSTALL( FILES platforms/linux/gemrb.desktop DESTINATION ${MENU_DIR} )
-		INSTALL( FILES platforms/linux/org.gemrb.gemrb.metainfo.xml DESTINATION ${METAINFO_DIR} )
-	ENDIF()
-	INSTALL( FILES README.md INSTALL COPYING NEWS AUTHORS DESTINATION ${DOC_DIR} )
-	INSTALL( FILES admin/extend2da.py DESTINATION ${BIN_DIR}
-		PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
-ENDIF()
-
-CONFIGURE_FILE(
-	"${CMAKE_CURRENT_SOURCE_DIR}/cmake/cmake_uninstall.cmake.in"
-	"${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
-	IMMEDIATE @ONLY
-)
-
-ADD_CUSTOM_TARGET( uninstall
-	"${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake" )
-
-add_custom_target(appimage
-    COMMAND rm -rf ./AppDir GemRB-*-x86_64.AppImage
-    COMMAND make install DESTDIR=./AppDir
-    COMMAND mv AppDir/usr/usr/share/gemrb AppDir/usr/share/ || true # not always needed
-    COMMAND sed -i 's,X-AppImage-Version.*,X-AppImage-Version='`date +%F`-$ENV{GITHUB_SHA}',' AppDir/usr/share/applications/gemrb.desktop
-    # bundle core python modules
-    # AppRun defaults PYTHONPATH to this destination
-    # source is probably available in Python3_STDLIB once we switch to newer cmake
-    COMMAND mkdir -p AppDir/usr/share/pyshared
-    COMMAND cp -r /usr/lib/python3.8/* AppDir/usr/share/pyshared
-    COMMAND rm -rf AppDir/usr/share/pyshared/config-*x86_64-linux-gnu
-    COMMAND rm -rf AppDir/usr/share/pyshared/dist-packages
-    COMMAND rm -rf AppDir/usr/share/pyshared/test
-    COMMAND rm -rf AppDir/usr/share/pyshared/pydoc_data
-    COMMAND rm -rf AppDir/usr/share/pyshared/lib2to3
-    COMMAND rm -rf AppDir/usr/share/pyshared/config-3*/libpython3*.a
-    COMMAND LD_LIBRARY_PATH=./AppDir/usr/lib/gemrb ./linuxdeploy --appdir=./AppDir --output=appimage --custom-apprun="${CMAKE_CURRENT_SOURCE_DIR}/platforms/linux/AppRun.sh"
-    WORKING_DIRECTORY ${CMAKE_BUILD_DIR}
-)
-
-# make dist for a gzipped tarball of current HEAD
-# include fetch_demo_data files manually, since they're not part of the repo
-set(ARCHIVE_NAME ${CMAKE_PROJECT_NAME}-${GEMRB_VERSION})
-add_custom_target( dist
-	COMMAND git archive --worktree-attributes --prefix=${ARCHIVE_NAME}/ -o ${ARCHIVE_NAME}.tar HEAD
-	COMMAND tar --append -f ${ARCHIVE_NAME}.tar --transform="s,^,${ARCHIVE_NAME}/," demo/music
-	COMMAND gzip --best -c ${ARCHIVE_NAME}.tar > ${CMAKE_BINARY_DIR}/${ARCHIVE_NAME}.tar.gz
-	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-)
-
-add_custom_target(fetch-demo-data
-	COMMAND ${CMAKE_COMMAND} -P ${CMAKE_SOURCE_DIR}/cmake/fetch_demo_data.cmake
-	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/demo
-)
-add_dependencies(dist fetch-demo-data)
-
-# pretty-print options macro
-# as of 2.8 cmake does not support anything like EVAL
-MACRO(PRINT_OPTION option)
-if (${option})
-	message(STATUS "  ${option}: ${${option}}")
-else()
-	message(STATUS "  ${option}: disabled")
-endif()
-ENDMACRO(PRINT_OPTION)
+CONFIGURE_COMPILER()
+INCLUDE(config)
```

Comment:
Isn't this now too early to be properly populated?

Reply:
Only this is needed

```
LIST(APPEND CMAKE_MODULE_PATH
	"${CMAKE_SOURCE_DIR}/cmake"
	"${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/"
)
```

Before ```INCLUDE(config)```.

Also
```CMAKE_CXX_COMPILER``` and ```CMAKE_C_COMPILER```
needs to be set before. I can made a sanity check for it.
I don't think it's needed since "everything" else will fail if you don't specify compilers at all.

Reply:
I'm asking because some defines are set based on cmake variables set later in the file. If the actual processing is deferred, then good, but do check your config.h looks as expected.

Reply:
I checked it also with comparing to the printed values and they're the same.

Reply:
We don't print things like SUPPORTS_MEMSTREAM.

Reply:
Yeah I know I meant only ones which are printed.
The rest of them were checked by comparing the `config.h` and the code itself.
Do you need some extra prints for it? (We can run this via CI and then I can remove them in subsequent commit).

Reply:
No, just making sure you tested fully.